core/src/main/scala/kafka/log/Log.scala

```diff
@@ -596,8 +600,8 @@ class Log(@volatile var dir: File,
       if (batchSize > config.maxMessageSize) {
         BrokerTopicStats.getBrokerTopicStats(topicPartition.topic).bytesRejectedRate.mark(records.sizeInBytes)
         BrokerTopicStats.getBrokerAllTopicsStats.bytesRejectedRate.mark(records.sizeInBytes)
-        throw new RecordTooLargeException(s"Message size is $batchSize bytes which exceeds the maximum configured " +
-          s"message size of ${config.maxMessageSize}.")
+        throw new RecordTooLargeException(s"The record batch size is $batchSize bytes which exceeds the maximum configured " +
```

Comment:
We're aware of this issue and documented it in the upgrade notes: https://github.com/apache/kafka/blob/trunk/docs/upgrade.html#L61

I'm not sure I'd call it a breaking change given the purpose of the config. In any case, I have it on my list to file an issue related to how we can evolve the configs so that their purpose is clearer (and to consider if we can drop some of them).